beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/DepositProvider.java

```diff
@@ -130,10 +138,27 @@ public void onNewFinalizedCheckpoint(
                 .getEth1DataAndHeight(finalizedState.getEth1Data())
                 .map(Eth1DataCache.Eth1DataAndHeight::getBlockHeight);
         if (heightOptional.isEmpty()) {
-          LOG.warn("Eth1Data height not found in cache. Skipping DepositTree pruning");
+          LOG.debug("Eth1Data height not found in cache. Skipping DepositTree pruning");
           return;
         }
         depositMerkleTree.finalize(finalizedState.getEth1Data(), heightOptional.get());
+        depositMerkleTree
+            .getSnapshot()
+            .ifPresent(
+                depositTreeSnapshot -> {
+                  LOG.debug("Storing DepositTreeSnapshot: {}", depositTreeSnapshot);
+                  storageUpdateChannel
+                      .onFinalizedDepositSnapshot(depositTreeSnapshot)
+                      .thenCompose(storeResult -> eth1DepositStorageChannel.removeDepositEvents())
+                      .exceptionally(
+                          throwable -> {
+                            LOG.error(
+                                "Failed to store snapshot and remove old deposit events",
+                                throwable);
+                            return null;
+                          })
+                      .ifExceptionGetsHereRaiseABug();
```

Comment:
You can just use `.finish(throwable ->` for this.

Reply:
Cool, used it